lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -97,6 +107,35 @@ public function __construct(mysqli $conn, $prepareString)
         $this->_bindedValues = array_fill(1, $paramCount, null);
     }
 
+    /**
+     * Converts named placeholders (":parameter") into positional ones ("?"), as MySQL does not support them.
+     *
+     * @param string $query The query string to create a prepared statement of.
+     *
+     * @return string
+     */
+    private function convertNamedToPositionalPlaceholders($query)
+    {
+        $numberOfCharsQueryIsShortenedBy = 0;
+        $placeholderNumber               = 0;
+
+        foreach (SQLParserUtils::getPlaceholderPositions($query, false) as $placeholderPosition => $placeholderName) {
+            $placeholderName = (string) $placeholderName;
+            if (array_key_exists($placeholderName, $this->placeholderNamesToNumbers) === false) {
+                $this->placeholderNamesToNumbers[$placeholderName] = [];
+            }
+
+            $this->placeholderNamesToNumbers[$placeholderName][] = $placeholderNumber++;
+
+            $placeholderPositionInShortenedQuery = $placeholderPosition - $numberOfCharsQueryIsShortenedBy;
+            $placeholderNameLength               = strlen($placeholderName);
+            $query                               = substr($query, 0, $placeholderPositionInShortenedQuery) . '?' . substr($query, ($placeholderPositionInShortenedQuery + $placeholderNameLength + 1));
```

Comment:
Let's use linebreaks and remove unneeded parenthesis
```suggestion
            $query                               = substr($query, 0, $placeholderPositionInShortenedQuery) . '?' . 
                substr($query, $placeholderPositionInShortenedQuery + $placeholderNameLength + 1);
```